edgemesh/pkg/proxy/proxy.go

```diff
@@ -544,11 +545,14 @@ func (c *conntrack) processServerProxy() {
 
 //isL4Proxy Determine whether to use L4 proxy
 func IsL4Proxy(svc *v1.Service) bool {
+	if svc.Spec.Ports == nil || len(svc.Spec.Ports) == 0 {
```

Comment:
Have `len(svc.Spec.Ports) == 0` this one is enough 
```suggestion
	if len(svc.Spec.Ports) == 0 {
```

Reply:
And switch suggest to:
```suggestion
	switch port.Name {
	case "websocket", "grpc", "https", "tcp":
		return true
	case "http", "udp":
		return false
	default:
		return true
	}
```

Reply:
Thanks for suggestions. please review again